p42/src/song.rs

```diff
@@ -0,0 +1,175 @@
+use core::panic;
+use std::io::{Read, Write};
+
+static DAYS: [&str; 12] = [
+    "first", "second", "third", "fourth", "fifth", "sixth", "seventh", "eighth", "ninth", "tenth",
+    "eleventh", "twelfth",
+];
+
+static GIFTS: [&str; 12] = [
+    " partridge in a pear tree",
+    "Two turtle doves,",
+    "Three French hens,",
+    "Four calling birds,",
+    "Five golden rings,",
+    "Six geese a-laying,",
+    "Seven swans a-swimming,",
+    "Eight maids a-milking,",
+    "Nine ladies dancing,",
+    "Ten lords a-leaping,",
+    "Eleven pipers piping,",
+    "Twelve drummers drumming,",
+];
+
+static LEFT: &str = "On the ";
+static RIGHT: &str = " day of Christmas,";
+static SECOND: &str = "my true love gave to me";
+static V1: &str = "A";
+static V2: &str = "And a";
+
+#[derive(Debug, Clone, Default, Copy)]
+pub struct SongIter {
+    pub day: usize,
+    pub line: usize,
```

Comment:
These fields can be made private and you can create the iterator using `SongIter::default()` or `SongIter::new()`.